src/main/flight/pid.c

```diff
@@ -349,9 +350,12 @@ static void pidApplyRateController(const pidProfile_t *pidProfile, pidState_t *p
 #endif
 }
 
-void updateMagHoldHeading(int16_t heading)
+void updateMagHoldHeading(int16_t heading, uint8_t reset)
```

Comment:
You are right, that was my initial thought too. But then came problem of flash and was not sure if new function will take more space or parameter will take more.

If that was the problem, we might rewrite it for 1.4 or even 1.3. If not then whole branch can be deleted :)